help-docs: Document "View user profile" mobile app feature.

drrosa opened the pull request:
Refactors the instructions for Desktop/Web and adds instructions for mobile app users.

**Screenshots and screen captures:**

<img width="536" alt="image" src="https://user-images.githubusercontent.com/2343554/178131484-33001311-bb15-460d-84bd-ab97e4deab01.png">

Consolidates the two ways of viewing a user profile into one set of instructions under a single Desktop/Web tab.

<img width="539" alt="image" src="https://user-images.githubusercontent.com/2343554/178131471-6ba3d728-aacd-4a7b-849e-48ff8265e919.png">

<details>
<summary>Self-review checklist</summary>

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [x] Automated tests

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/version-control.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

drrosa commented:
@alya ready for review.

alya commented on templates/zerver/help/view-someones-profile.md:
```diff
@@ -10,21 +10,16 @@ streams for which [you can see all subscribers](/help/stream-permissions).
 
-{!right-sidebar-view-full-profile.md!}
+    You can also click on a user's profile picture or name and then
+    click **View full profile**.
```
I wonder if we could put this tip above the "View full profile" instruction to be able to drop the "View full profile" part of it? Can we try that and see if it looks confusing?

drrosa replied:
That's a good idea, but unfortunately, it would break up the numbering i.e. the number in step three would get rendered as a number `1` instead of `3`.

alya commented on templates/zerver/help/view-someones-profile.md:
```diff
@@ -10,21 +10,16 @@ streams for which [you can see all subscribers](/help/stream-permissions).
+!!! Tip ""
 
-{!right-sidebar-view-full-profile.md!}
+    You can also click on a user's profile picture or name and then
```
Maybe "... picture or name in the message list", since the reference to the message list was previously contained in the heading?

alya replied:
How about something like this for the tip? I think it's helpful to be a bit more specific.

> You can also click on a user's profile picture or name on a message they sent, and skip to the last step.

I like the bottom placement of the tip, since it makes it easier to follow the instructions without getting distracted by it if you don't need it.

alya commented on templates/zerver/help/view-someones-profile.md:
```diff
@@ -21,6 +21,15 @@ streams for which [you can see all subscribers](/help/stream-permissions).
+!!! Tip ""
+
+    If you are viewing a private message, you can also tap the **ⓘ** icon
+    in the upper right of the screen.
```
Maybe "upper right corner of the screen"?

alya commented on templates/zerver/help/view-someones-profile.md:
```diff
@@ -21,6 +21,15 @@ streams for which [you can see all subscribers](/help/stream-permissions).
+
+!!! Tip ""
+
+    If you are viewing a private message, you can also tap the **ⓘ** icon
```
Is there a more robust way to put in the same icon as we have in the mobile app? It's not a huge deal, but this doesn't look quite right.

drrosa replied:
It seems like the exact same icon is not available in this codebase, but maybe we could pull it from zulip-mobile?

We could use a similar one that's available though:
<img width="531" alt="image" src="https://user-images.githubusercontent.com/2343554/178609431-312d2784-e018-447d-b36d-8e6a0c7545a1.png">

alya replied:
Let's try to figure our how to use the correct icon, not one that's somewhat similar. If you aren't sure how best to do it from a technical perspective, could you start a thread on CZO?

drrosa replied:
Sounds good!
https://chat.zulip.org/#narrow/stream/19-documentation/topic/need.20mobile.20icons.20for.20docs

drrosa commented:
I've pushed the requested changes to [View someone's profile](https://zulip.com/help/view-someones-profile). I also started a [discussion thread](https://chat.zulip.org/#narrow/stream/19-documentation/topic/need.20mobile.20icons.20for.20docs) to figure out how to get the correct information **ⓘ** icon, but maybe we can merge this PR and do that as a follow-up.

<img width="527" alt="image" src="https://user-images.githubusercontent.com/2343554/179867816-9538c23b-f251-43bb-8dfd-095690ce5cff.png">

<img width="524" alt="image" src="https://user-images.githubusercontent.com/2343554/179867858-02640976-d8df-4cd1-8940-7057f301a6bc.png">

alya commented:
Let's put the icon in parentheses (cf. https://zulip.readthedocs.io/en/latest/documentation/helpcenter.html#icons). Otherwise looking good!

drrosa commented:
> Let's put the icon in parentheses

Done!

<img width="527" alt="image" src="https://user-images.githubusercontent.com/2343554/180700063-2ca60636-e55b-49d1-a265-5db00423a663.png">

alya commented:
Thanks! @timabbott please take a look.

drrosa commented:
> I've pushed the requested changes to [View someone's profile](https://zulip.com/help/view-someones-profile). I also started a [discussion thread](https://chat.zulip.org/#narrow/stream/19-documentation/topic/need.20mobile.20icons.20for.20docs) to figure out how to get the correct information **ⓘ** icon, but maybe we can merge this PR and do that as a follow-up.

Fixed the icon using the same strategy as in https://github.com/zulip/zulip/pull/22447#issuecomment-1216111114.


<img width="533" alt="image" src="https://user-images.githubusercontent.com/2343554/185817465-f5201d5a-b971-4a15-875b-5d428685e118.png">

Also fixed the icon in https://zulip.com/help/pin-a-stream#alternate-method as a follow-up to #22429.

<img width="536" alt="image" src="https://user-images.githubusercontent.com/2343554/185817418-dd5e660b-61ef-43b3-ab2a-42c63239000b.png">

alya commented:
@timabbott I reviewed this one before my trip, and @drrosa has addressed the minor feedback I had, as well as updating the icon we're using. Could you please take a look?

timabbott commented on templates/zerver/help/view-someones-profile.md:
```diff
@@ -18,8 +18,18 @@ streams for which [you can see all subscribers](/help/stream-permissions).
+
+!!! Tip ""
+
+    If you are viewing a private message, you can also tap the information
```
This should be "viewing a private message conversation,", right? We don't have views that show just individual messages.

drrosa replied:
Yes, good catch!

drrosa commented:
@timabbott Fixed it. Thanks!

timabbott commented:
Merged, thanks @drrosa!
